### PR TITLE
add go dockerfile test and no modules

### PR DESCRIPTION
**Pull request by davidgamero**
# Description

add the non-module supporting `go` dockerfile support to an integration test

## Type of change

Please delete options that are not relevant.

- [x] New features

# How Has This Been Tested?

- [x] New integration tests for `go` without modules

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by bfoley13 — APPROVED** [review]

File: test/integration_config.json [hunk 1]
```diff
@@ -6,6 +6,13 @@
     "serviceport": 80,
     "repo": "gambtho/go_echo"
   },
+  {
+    "language": "go",
+    "version": "1.20",
+    "port": "8080",
+    "serviceport": 8080,
+    "repo": "davidgamero/go-echo-no-mod"
```
Comment (Tatsinnit) [inline test/integration_config.json:14]: 💡 fyi test had your alias - is this meant to be that way or placeholder for user alias et. al. Thanks heaps.

Comment (Tatsinnit) [reply test/integration_config.json:14]: 🧙  but then I see other example with other user alias so all good, jsut fyi.

**Review by Tatsinnit — APPROVED** [review]
👍 ❤️

File: .github/workflows/integration-linux.yml [hunk 2]
```diff
@@ -292,6 +292,270 @@ jobs:
...
 
+  go-helm-dry-run:
+      runs-on: ubuntu-latest
+      needs: build
+      steps:
+        - uses: actions/checkout@v3
+        - uses: actions/download-artifact@v3
+          with:
+            name: draft-binary
+        - run: chmod +x ./draft
```
Comment (Tatsinnit) [inline .github/workflows/integration-linux.yml:303]: 💡 Note: `+x` for all? as default could be too much between user, group and others, just an fyi. 

Recommendation: `755` could be better but I am not sure what context will this run. hence sharing my thoughts.

File: test/integration_config.json (hunk 1 above)
Comment (davidgamero) [reply test/integration_config.json:14]: yep the integration tests pull use those repos to run draft against for validating it works on each language

**Pull request merged**